src/main/java/cat/udl/eps/softarch/unicloud/domain/Degree.java

```diff
@@ -0,0 +1,48 @@
+package cat.udl.eps.softarch.unicloud.domain;
+
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+
+@Entity
+@Table
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class Degree extends UriEntity<Long> {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    @NotBlank
+    @Column(nullable = false)
+    private String name;
+
+    @NotBlank
+    @Column(nullable = false)
+    private String faculty;
+
+    @Override
+    public Long getId() {
```

Comment:
Using Lombok with the annotation @Data all setters and getters are implemented automatically, so you don't need to define them explicitly in your code: https://projectlombok.org/features/Data

You can delete the getters and setters in this class.